camus-etl-kafka/src/main/java/com/linkedin/camus/etl/kafka/CamusJob.java

```diff
@@ -390,6 +390,10 @@ public int compare(FileStatus f1, FileStatus f2) {
         && props.getProperty(ETL_FAIL_ON_ERRORS, Boolean.FALSE.toString()).equalsIgnoreCase(Boolean.TRUE.toString())) {
       throw new RuntimeException("Camus saw errors, check stderr");
     }
+
+    if (EtlInputFormat.reportJobFailureDueToOffsetOutOfRange) {
```

Comment:
It's also better to introduce a config parameter to decide whether to throw Exception on this case.

Also How did you test all these error cases?